private/bufpkg/bufmodule/bufmodulecache/module_cacher.go

```diff
@@ -142,6 +139,18 @@ func (m *moduleCacher) PutModule(
 	if err := bufmodule.ModuleToBucket(ctx, module, dataReadWriteBucket); err != nil {
 		return err
 	}
+	bucketModule, err := bufmodule.NewModuleForBucket(
+		ctx,
+		dataReadWriteBucket,
+		bufmodule.ModuleWithModuleIdentityAndCommit(modulePin, modulePin.Commit()),
+	)
+	if err != nil {
+		return fmt.Errorf("failed to write module to bucket: %w", err)
+	}
+	digest, err := bufmodule.ModuleDigestB3(ctx, bucketModule)
+	if err != nil {
+		return err
+	}
```

Comment:
Yeah, I think that's a reasonable pushback. Let me take a look at populating the default config for the returned proto structure. In that case, we likely need to file an issue to clean-up some of the configuration package so that the defaulted behaviour is less spread out and more clear across the package.